ci-operator/config/openshift/hive/master.yaml

```diff
@@ -0,0 +1,61 @@
+base_images:
+  base:
+    cluster: https://api.ci.openshift.org
+    name: origin-v4.0
+    namespace: openshift
+    tag: base
+  root:
+    cluster: https://api.ci.openshift.org
+    name: release
+    namespace: openshift
+    tag: golang-1.10
+binary_build_commands: make build
+canonical_go_repository: github.com/openshift/hive
+images:
+- dockerfile_path: build/hive/Dockerfile
+  from: base
+  inputs:
+    bin:
+      paths:
+      - destination_dir: .
+        source_path: /go/src/github.com/openshift/hive/bin/manager
+  to: hive
+- dockerfile_path: build/build-image/Dockerfile
+  from: root
+  optional: true
+  to: builder-base
+- dockerfile_path: build/test-image/Dockerfile
+  from: builder-base
+  optional: true
+  to: test-base
+resources:
+  '*':
+    limits:
+      cpu: '2'
+      memory: 4Gi
+    requests:
+      cpu: 100m
+      memory: 200Mi
+  unit:
+    limits:
+      cpu: '4'
+      memory: 4Gi
+    requests:
+      cpu: '2'
+      memory: 1Gi
+tag_specification:
+  cluster: https://api.ci.openshift.org
+  name: origin-v4.0
+  namespace: openshift
+  tag: ''
+  tag_overrides: {}
+build_root:
+  image_stream_tag:
+    cluster: https://api.ci.openshift.org
+    name: release
+    namespace: openshift
+    tag: golang-1.10
+tests:
+- as: unit
+  commands: make test
+  from: test-base
```

Comment:
So you should be able to get away with:

```yaml
build_root:
  project_image:
    dockerfile_path: build/test-image/Dockerfile
tests:
- as: unit
  commands: make test
  from: src
```

and drop the `RUN mkdir -p` and the `WORKDIR` from the builder-base, drop the test-base entirely

Reply:
Hey Steve, thanks for the review.

When I make the changes you suggest and run ci-operator locally (0771d83 on master branch), I get this error:
```
[twiest@vm hive (add-hive-to-ci *%=)]$ ci-operator --config  $HOME/code/release/ci-operator/config/openshift/hive/master.yaml --target unit --git-ref "twiest/hive@ci-changes"
error: invalid configuration: you have to specify either git_source_image or image_stream_tag for the build_root
```

I don't see git_source_image in the [ci-operator documentation](https://github.com/openshift/ci-operator/blob/master/CONFIGURATION.md#build_root), so I'm not sure how to proceed.

Here's my `build_root` and `tests` sections:
```yaml
build_root:
  project_image:
    dockerfile_path: build/build-image/Dockerfile
tests:
- as: unit
  commands: make test
  from: src
```

Reply:
Oops, we had an issue with that config validation check. I've fixed that, can you try again with a fresh pull of the `ci-operator` image or a re-build from `master`?